src/util/Collection.js

```diff
@@ -197,16 +197,20 @@ class Collection extends Map {
   /**
    * Removes entries that satisfy the provided filter function.
    * @param {Function} fn Function used to test (should return a boolean)
-   * @param {Object} [thisArg] Value to use as `this` when executing function
-   * @returns {number} The number of removed entries
+   * @param {*} [thisArg] Value to use as `this` when executing function
+   * @returns {Collection} The sweeped entries
+   * @example collection.sweep(message => message.deleted);
    */
   sweep(fn, thisArg) {
     if (typeof thisArg !== 'undefined') fn = fn.bind(thisArg);
-    const previousSize = this.size;
+    const results = new Collection();
```

Comment:
the point of sweep is to save memory. there is no reason for this change. hell nah.